platform-service/src/unstract/platform_service/helper.py

```diff
@@ -109,12 +109,19 @@ def __init__(
         self.file_path = file_path
         self.logger = logger
 
-    def calculate_cost(self, model_name, input_tokens, output_tokens):
+    def calculate_cost(self, model_name, provider, input_tokens, output_tokens):
         cost = 0.0
         item = None
         model_prices = self._get_model_prices()
-        if model_prices:
-            item = model_prices.get(model_name)
+        # Filter the model objects by model name
+        filtered_models = {
+            k: v for k, v in model_prices.items() if k.endswith(model_name)
```

Comment:
@Deepak-Kesavan why do we check with `k.endswith(model_name` instead of doing a direct match?

Reply:
If the same model is provided by multiple LLM providers, then the pattern will be like `provider/model-name`. That's why